sympy/integrals/rubi/Algebra_Integration_rules.py

```diff
@@ -0,0 +1,40 @@
+# File: SineIntegrationRules.py
+#
+# This file is a Python code generated from the publicly available rule-based
+# integrator (Rubi) system created by Albert Rich available at
+# http://www.apmaths.uwo.ca/~arich/. This file is under the modified BSD
+# #TODO: license (see the LICENSE file for more details).
+
+from sympy import simplify, log
+from sympy.core.symbol import Symbol, symbols
+from sympy.core.symbol import Wild
+
+def rubi_integrate(self, x):
+	"""
+	rubi_integrate decides which category of rule does the
+	given integrand (self) falls in and returns to that rule.
+	"""
+    if self.is_Mul:
+        if self.args[0].has(x) == False and len(self.args) == 2:
+            # condition to check if the first arg is a constant,
+            # returns in an is_Pow form so that it matches better.
+            return (self.args[0]*int_alg_1_1(self.args[1], x))
+    if self.is_Pow:
+        return (int_alg_1_1(self, x))
+
+def int_alg_1_1(self, x):
+    # rule int_alg_1_1 indicates Algebraic > Linear Product > (a + b*x)**m category
+    a = Wild('a', exclude = [x])
+    b = Wild('b', exclude = [x])
+    m = Wild('m', exclude = [x])
+    simplify(self)
+    type = self.match_between(((b*x)**m), (a + b*x)**m)
+    b = type[b]
+    m = type[m]
+    a = type[a]
```

Comment:
I guess, it won't fail, `match_between()` will give the following result you may see test cases.  
```
(a + b*x)**m    (matched output)
(b*x)**m        (a_ = 0, b_ = b, m_= m)
(a + x)**m     (a_ = 0, b_ = 1, m_= m)
a + b*x         (a_ = 0, b_ = b, m_= 1) # I doubt this case current
 match would not give correct match.
```
it would return maximum possible variable match and would help avoiding such cases where we need to handle variables individually, `a = type[a]` will never end up with `KeyError`.
see these test cases:
```
    assert int_alg_1_1((5 + 6*x)**3, x) == (6*x + 5)**4/24
    assert rubi_integrate((5 + 6*x)**3, x) == (6*x + 5)**4/24
    assert rubi_integrate((6*x)**3, x) == 54*x**4
    assert rubi_integrate(2*(2 + 3*x)**5, x) == (3*x + 2)**6/9
```
In all cases I think it matches with `(a +b*x)**m`, if I'm not wrong.